### ISSUE TITLE
Feature request: Step/Shake/Tap tracking

### ISSUE THREAD

**Issue by sajattack** [NeoBirth/accelerometer.rs#57]
Hi @tarcieri,

I'm planning to write an accelerometer.rs driver for the pinetime smartwatch, and it would be nice if there were more types of tracking available than just orientation. Do you have any plans to implement these?

**Comment by tarcieri** [comment]
I'd definitely be interested in adding them, particularly if you have API suggestions.

It seems like these could potentially have their own traits with a simple API like:

```rust
pub trait Tap {
    fn is_tapped(&mut self) -> bool;
}
```

...although I'm definitely open to suggestions.

**Comment by sajattack** [comment]
I was thinking of something similar to the Orientation API, but actually now I see this accelerometer has most of these features implemented in hardware. Still, for other accelerometers, it would be good to have a software implementation.

**Comment by tarcieri** [comment]
Yeah, I proposed that API because my accelerometer also has shake/tap detection in hardware.

I think exposing that through a common API is probably the right place to start, and then perhaps adding a software implementation for accelerometers which don't have hardware support. That seems a bit trickier, though.
